lib/liquid/context.rb

```diff
@@ -74,32 +74,13 @@ def pop_interrupt
     end
 
     def handle_error(e, line_number = nil, raw_token = nil)
-      if e.is_a?(Liquid::Error)
-        e.template_name ||= template_name
-        e.line_number ||= line_number
-      end
-
-      output = nil
-
-      if exception_handler
-        args = [e]
-        args << { line_number: line_number, raw_token: raw_token } if exception_handler.arity == 2
```

Comment:
Hmm is this logic not necessary anymore? Was this only a hack to get the line number and now you are attaching it to the exception itself (since everything is a `Liquid::Error` or `Liquid::InternalError` now)?